src/main.rs

```diff
@@ -204,28 +204,38 @@ fn setup(args: &[String]) -> Setup {
     let mixer_name = matches.opt_str("mixer");
     let mixer = mixer::find(mixer_name.as_ref()).expect("Invalid mixer");
 
+    let use_audio_cache = !matches.opt_present("disable-audio-cache");
+
+    let cache = matches
+        .opt_str("c")
+        .map(|cache_location| Cache::new(PathBuf::from(cache_location), use_audio_cache));
+
+    let mut default_volume: u16 = 0x8000;
+    let cached_volume = cache.as_ref().and_then(Cache::volume);
+
+    // override default volume with cached volume if found
+    if cached_volume.is_some() {
+        default_volume = cached_volume.unwrap().volume;
+    }
+
+    // override default/cached volume with initial volume if found
     let initial_volume = matches
         .opt_str("initial-volume")
         .map(|volume| {
-            let volume = volume.parse::<i32>().unwrap();
-            if volume < 0 || volume > 100 {
+            let volume = volume.parse::<u16>().unwrap();
+            if volume > 100 {
                 panic!("Initial volume must be in the range 0-100");
             }
-            volume * 0xFFFF / 100
+            (volume as i32 * 0xFFFF / 100) as u16
         })
-        .unwrap_or(0x8000);
+        .unwrap_or(default_volume);
```

Comment:
Untested, but this should work and remove the need for the `default_volume` and `cached_volume` temporaries.
```rust
let initial_volume = matches
     .opt_str("initial-volume")
     .map(|volume| {
        let volume = volume.parse::<u16>().unwrap();
        if volume > 100 {
            panic!("Initial volume must be in the range 0-100");
        }
        (volume as i32 * 0xFFFF / 100) as u16
     })
     .or_else(|| cache.as_ref().and_then(Cache::volume));
     .unwrap_or(0x8000);
```